x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityTests.java

```diff
@@ -514,6 +525,44 @@ public void testLicenseUpdateFailureHandlerUpdate() throws Exception {
         }
     }
 
+    public void testSecurityHandlerIsAlwaysInstalled() {
+        Settings settings = Settings.builder()
+            .put("xpack.security.enabled", false)
+            .put("path.home", createTempDir())
+            .build();
+
+        XPackPlugin xpackPlugin = new LocalStateCompositeXPackPlugin(settings, null);
+        SettingsModule settingsModule = new SettingsModule(Settings.EMPTY);
+        ThreadPool threadPool = new TestThreadPool(getTestName());
+
+        try {
+            UsageService usageService = new UsageService();
+            Security security = new Security(settings, null);
+
+            ActionModule actionModule = new ActionModule(settingsModule.getSettings(),
+                TestIndexNameExpressionResolver.newInstance(threadPool.getThreadContext()),
+                settingsModule.getIndexScopedSettings(), settingsModule.getClusterSettings(), settingsModule.getSettingsFilter(),
+                threadPool, Arrays.asList(xpackPlugin, security), null, null, usageService, null);
+            actionModule.initRestHandlers(null);
+
+            // At this point the easiest way to confirm that a handler is loaded is to try to register another one on top of it and to fail
```

Comment:
This does not really test that the rest wrapper is installed and is from the Security plugin. It only tests that a rest handler for path `/` is installed, which happens regardless with or without current changes.

In my opinion, a simple unit test for `Security#getRestHandlerWrapper` is sufficient, i.e. simply test the handler wrapper is returned regardless of the value of `xpack.security.enabled`.

If you prefer the current approach, it might be possible to register a new handler and try to invoke it. Along the way we can try inspect the call stack to ensure the security rest wrapper is in action. But in general, I feel testing at this level is not much, if at all, better than the unit test. If we do want to test the real thing, the same approach of inspecting call stack is probably better suited for an external cluster test.

Reply:
This test tests that despite `xpack.security.enabled` being set to `false`, the Rest Handler has been installed. But I, actually, added one more test which ensures the second plugin trying to install a rest handler will fail.

Reply:
Aren't we trying to test whether the rest **wrapper** is installed instead of the rest "handler"?

Reply:
I think one implies another. There is no easy way to access the Wrapper through the rest controller, but we can tell the rest wrapper was installed by initializing rest handlers and trying to register another one.

Reply:
> I think one implies another.

I disagree. This test passes even when no rest Wrapper is installed. That is, if you replace `Arrays.asList(xpackPlugin, security)` with `List.of()` in the constructor call of `ActionModule`, the test passes by throwing the same error.

Reply:
You are right! Changed the test to look at the debug message instead